pkg/operator/context/context.go

```diff
@@ -69,14 +69,11 @@ func New(
 	}
 	ctx.APIs = apis
 
-	for _, api := range ctx.APIs {
-		if api.RequestHandler != nil {
-			ctx.ProjectID = projectID
-			ctx.ProjectKey = filepath.Join(consts.ProjectsDir, ctx.ProjectID+".zip")
-			if err = config.AWS.UploadBytesToS3(projectBytes, ctx.ProjectKey); err != nil {
-				return nil, err
-			}
-			break
+	if userconf.APIs.AreProjectFilesRequired() {
```

Comment:
Just a thought, totally up to you: We could add a top-level `userconf.AreProjectFilesRequired()` (which just calls `userconf.APIs.AreProjectFilesRequired()`, or loops through the APIs and then delete `userconf.APIs.AreProjectFilesRequired()`) if you think it makes sense at that scope